src/lib/interactionManager/interaction.ts

```diff
@@ -0,0 +1,244 @@
+  public async createAuthenticationResponse() {
+    // TODO Abstract to getMessages * findByType
+    return this.ctx.identityWallet.create.interactionTokens.response.auth(
+      this.getSummary().state,
+      await this.ctx.keyChainLib.getPassword(),
+      this.getMessages().find(
+        ({ interactionType }) =>
+          interactionType === InteractionType.Authentication,
+      ),
+    )
+  }
+
+  public async createCredentialResponse(
+    selectedCredentials: CredentialVerificationSummary[],
+  ) {
+    // TODO Abstract to getMessages * findByType
+    const request = this.getMessages().find(
+      ({ interactionType }) =>
+        interactionType === InteractionType.CredentialRequest,
+    ) as JSONWebToken<CredentialRequest>
+
+    const credentials = await Promise.all(
+      selectedCredentials.map(
+        async ({ id }) => (await this.getVerifiableCredential({ id }))[0],
+      ),
+    )
+
+    return this.ctx.identityWallet.create.interactionTokens.response.share(
+      {
+        callbackURL: request.interactionToken.callbackURL,
+        suppliedCredentials: credentials.map(c => c.toJSON()),
+      },
+      await this.ctx.keyChainLib.getPassword(),
+      request,
+    )
+  }
+
+  public async createCredentialOfferResponseToken(
+    selectedOffering: SignedCredentialWithMetadata[],
+  ) {
+    const credentialOfferRequest = this.getMessages().find(
+      ({ interactionType }) =>
+        interactionType === InteractionType.CredentialOfferRequest,
+    ) as JSONWebToken<CredentialOfferRequest>
+
+    const credentialOfferResponseAttr = {
+      callbackURL: credentialOfferRequest.interactionToken.callbackURL,
+      selectedCredentials: selectedOffering.map(offer => ({
+        type: offer.type,
+      })),
+    }
+
+    return this.ctx.identityWallet.create.interactionTokens.response.offer(
+      credentialOfferResponseAttr,
+      await this.ctx.keyChainLib.getPassword(),
+      credentialOfferRequest,
+    )
+  }
+
+  // rename to signal this does validation
+  public async processInteractionToken(token: JSONWebToken<JWTEncodable>) {
+    // At some point we should strip the JSONWebToken<JWTEncodable>
+    if (!this.issuerSummary) {
+      // TODO Potential bug if we start with our token, i.e. we are the issuer
+      this.issuerSummary = generateIdentitySummary(
+        await this.ctx.registry.resolve(token.signer.did),
+      )
+    }
+
+    if (!this.flow) {
+      this.flow = new this.interactionFlow[token.interactionType](this)
+    }
+
+    if (token.signer.did !== this.ctx.identityWallet.did) {
+      await this.ctx.identityWallet.validateJWT(
+        token,
+        last(this.getMessages()),
+        this.ctx.registry,
+      )
+    }
+
+    await this.ctx.identityWallet.validateJWT(
+      token,
+      undefined, // TODO Extract from .getMessages()
+      this.ctx.registry,
+    )
+
+    if (token.interactionType === InteractionType.CredentialsReceive) {
```

Comment:
This feels hacky, but not sure where to move to.

Reply:
Does it fit into the flow handler?

Reply:
+1 For now I think that's a better place than on the `Interaction` class.